examples/calibration/plot_calibration_multiclass.py

```diff
@@ -3,83 +3,112 @@
 
-import numpy as np
+# * train: 600 samples (for training the classifier)
+# * valid: 400 samples (for calibrating predicted probabilities)
+# * test: 1000 samples
+
+# Note that we also create `X_train_valid` and `y_train_valid`, which consists
+# of both the train and valid subsets. This is used when we only want to train
+# the classifier but not calibrate the predicted probabilities.
 
+import numpy as np
 from sklearn.datasets import make_blobs
-from sklearn.ensemble import RandomForestClassifier
-from sklearn.calibration import CalibratedClassifierCV
-from sklearn.metrics import log_loss
 
 np.random.seed(0)
 
-# Generate data
-X, y = make_blobs(n_samples=1000, random_state=42, cluster_std=5.0)
+X, y = make_blobs(n_samples=2000, n_features=2, centers=3, random_state=42,
+                  cluster_std=5.0)
 X_train, y_train = X[:600], y[:600]
-X_valid, y_valid = X[600:800], y[600:800]
-X_train_valid, y_train_valid = X[:800], y[:800]
-X_test, y_test = X[800:], y[800:]
+X_valid, y_valid = X[600:1000], y[600:1000]
+X_train_valid, y_train_valid = X[:1000], y[:1000]
+X_test, y_test = X[1000:], y[1000:]
+
+# %%
+# Fitting and calibration
+# -----------------------
+#
+# First, we will train a :class:`~sklearn.ensemble.RandomForestClassifier`
+# with 25 base estimators (trees) on the concatenated train and validation
+# data (1000 samples). This is the uncalibrated classifier.
+
+from sklearn.ensemble import RandomForestClassifier
 
-# Train uncalibrated random forest classifier on whole train and validation
-# data and evaluate on test data
 clf = RandomForestClassifier(n_estimators=25)
 clf.fit(X_train_valid, y_train_valid)
-clf_probs = clf.predict_proba(X_test)
-score = log_loss(y_test, clf_probs)
 
-# Train random forest classifier, calibrate on validation data and evaluate
-# on test data
+# %%
+# To train the calibrated classifier, we start with the same
+# :class:`~sklearn.ensemble.RandomForestClassifier` but train it using only
+# the train data subset (600 samples) then calibrate, with `method='sigmoid'`,
+# using the valid data subset (400 samples) in a 2-stage process.
+
+from sklearn.calibration import CalibratedClassifierCV
+
 clf = RandomForestClassifier(n_estimators=25)
 clf.fit(X_train, y_train)
-clf_probs = clf.predict_proba(X_test)
-sig_clf = CalibratedClassifierCV(clf, method="sigmoid", cv="prefit")
-sig_clf.fit(X_valid, y_valid)
-sig_clf_probs = sig_clf.predict_proba(X_test)
-sig_score = log_loss(y_test, sig_clf_probs)
+cal_clf = CalibratedClassifierCV(clf, method="sigmoid", cv="prefit")
+cal_clf.fit(X_valid, y_valid)
+
+# %%
+# Compare probabilities
+# ---------------------
+#
+# The uncalibrated classifier is overly confident in its predictions and
+# incurs a large :ref:`log loss <log_loss>`. The calibrated classifier produces
+# probabilities that are less confident, i.e., the probabilities are further
+# away from 0 and 1. This calibration results in a lower log loss.
```

Comment:
Maybe we should remove this paragraph or expand the one just before or just after the code snippets that compares the log loss values.

Reply:
Good idea, amended.